src/pykeen/metrics/ranking.py

```diff
@@ -810,6 +855,30 @@ class ArithmeticMeanRank(RankBasedMetric):
                        &= \frac{1}{n^2} \sum \limits_{i=1}^{n} \frac{N_i^2 - 1}{12} \\
                        &= \frac{1}{12 n^2} \cdot \left(-n + \sum \limits_{i=1}^{n} N_i \right)
 
+    **Weighted Case**
+
+    When weights $w_1, \ldots, w_n$ are provided, the weighted mean rank and its moments
+    are:
+
+    .. math::
+
+        \text{Weighted MR} = \frac{\sum_{i=1}^{n} w_i r_i}{\sum_{j=1}^{n} w_j}
+
+    The expected value is:
+
+    .. math::
+
+        \mathbb{E}[\text{Weighted MR}] = \frac{\sum_{i=1}^{n} w_i \mathbb{E}[r_i]}{\sum_{j=1}^{n} w_j}
```

Comment:
it's not clear why you're able to move the expectation operator inside the sum where you do. The improvements to the docs in utils.py make this more clear - can we link to that? or explicitly copy it here too for completeness/self-contained-ness